### PR TITLE
LPS-182384 move portal-upload to modules part One

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina,

This is the clean up for directly using UploadServletRequestImpl() and UploadPortletRequestImpl() in the production code and test.
Thanks for checking

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:echo

**Comment by jeffwu0724** [comment]
ci:test:lima

**Comment by jeffwu0724** [comment]
Seems like this SF issue was reported before https://issues.liferay.com/browse/LPS-187249, and it is not caused by this ticket.
And if I run "ant format-source-current-branch", it will run successfully.
Thanks for checking!

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Comment by jeffwu0724** [comment]
ci:test:echo

**Comment by jeffwu0724** [comment]
ci:test:lima

**Comment by Kvale1733** [comment]
@jeffwu0724 Regression found


com.liferay.info.request.struts.test.EditInfoItemStrutsActionTest
```
 [exec] com.liferay.info.request.struts.test.EditInfoItemStrutsActionTest > testEditInfoItemInvalidBigDecimalTooBig STARTED
[beanshell] 2023-07-07 02:15:26.673 WARN  [com.liferay.info.test-executor-thread][UploadServletRequestImpl:192] Unable to parse upload request: the request was rejected because no multipart boundary was found
     [exec] 
     [exec] com.liferay.info.request.struts.test.EditInfoItemStrutsActionTest > testEditInfoItemInvalidBigDecimalTooBig FAILED
     [exec]     com.liferay.portal.kernel.events.ActionException: java.lang.NullPointerException
     [exec]     	at com.liferay.portal.events.ServicePreAction.run(ServicePreAction.java:161)
     [exec]     	at com.liferay.portal.kernel.events.Action.processLifecycleEvent(Action.java:34)
     [exec]     	at com.liferay.portal.events.EventsProcessorUtil.process(EventsProcessorUtil.java:80)
     [exec]     	at com.liferay.portal.events.EventsProcessorUtil.process(EventsProcessorUtil.java:52)
     [exec]     	at com.liferay.info.request.struts.test.EditInfoItemStrutsActionTest._processEvents(EditInfoItemStrutsActionTest.java:443)
```

```
    [exec]     Caused by: java.lang.NullPointerException
     [exec]     	at com.liferay.portal.upload.UploadServletRequestImpl.getParameterNames(UploadServletRequestImpl.java:491)
     [exec]     	at com.liferay.portal.upload.UploadPortletRequestImpl.getParameterNames(UploadPortletRequestImpl.java:251)
     [exec]     	at com.liferay.portal.upload.UploadPortletRequestImpl.getParameterMap(UploadPortletRequestImpl.java:235)
     [exec]     	at com.liferay.portal.kernel.service.ServiceContextFactory._getInstance(ServiceContextFactory.java:251)
     [exec]     	at com.liferay.portal.kernel.service.ServiceContextFactory.getInstance(ServiceContextFactory.java:63)
     [exec]     	at com.liferay.portal.events.ServicePreAction.servicePre(ServicePreAction.java:190)
     [exec]     	at com.liferay.portal.events.ServicePreAction.run(ServicePreAction.java:158)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-07/test-1-37/test-portal-acceptance-pullrequest(master)/13005/modules-integration-mysql57-jdk8/0/11/jenkins-console.txt.gz?authuser=0)

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Comment by jeffwu0724** [comment]
ci:test:echo

**Comment by jeffwu0724** [comment]
ci:test:lima

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

File: modules/apps/portal/portal-upload-test-util/src/main/java/com/liferay/portal/upload/test/util/UploadTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import javax.portlet.PortletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Jiefeng Wu
+ */
+public class UploadTestUtil {
+
+	public static UploadPortletRequest createUploadPortletRequest(
```
Comment (jeffwu0724) [inline modules/apps/portal/portal-upload-test-util/src/main/java/com/liferay/portal/upload/test/util/UploadTestUtil.java:36]: The reason I added this method is, I would like to centralize the logic of using new UploadPortletRequestImpl() in the test.
Later when I moved UploadPortletRequestImpl() to modules, I will use BundleWiring Bundles  to get the logic of Impl as Dante advised, and I will replace new UploadPortletRequestImpl() to using Bundle in this code

**Comment by jeffwu0724** [comment]
fix the conflict, need to retrigger the test

Hey @Kvale1733 , could you please help me double check the result when it is ready?
Thanks!

**Comment by jeffwu0724** [comment]
ci:test:relevant

**Comment by jeffwu0724** [comment]
ci:test:echo

**Comment by jeffwu0724** [comment]
ci:test:lima

**Comment by jeffwu0724** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@jeffwu0724 No regressions found.

**Comment by jeffwu0724** [comment]
updated commit message to "LPS-182384 added new method UploadTestUtil.createUploadPortletRequest(), to avoid directly calling UploadPortletRequestImpl() in the usages. As a preparation to move UploadPortletRequestImpl() to portal-upload-impl/internal, I would like to centralize the usages of UploadPortletRequestImpl() in the test. Later, I will use BundleWiring Bundles to get the logic of Impl from internal"

**Pull request closed without merging**